OpenSim/Tools/IMUInverseKinematicsTool.cpp

```diff
@@ -211,35 +211,45 @@ void IMUInverseKinematicsTool::runInverseKinematicsWithOrientationsFromFile(
+    if (resultsDir.empty() && !get_output_motion_file().empty())
+        resultsDir = IO::getParentDirectory(get_output_motion_file());
+    if (!resultsDir.empty()) {
+        IO::makeDir(resultsDir);
+        // directory will be restored on block exit
+        // by changing dir all other files are created in resultsDir
+        auto cwd = IO::CwdChanger::changeTo(resultsDir);
+        std::string outName = get_output_motion_file();
+        outName = IO::GetFileNameFromURI(outName);
+        if (outName.empty()) {
+            bool isAbsolutePath;
+            string directory, fileName, extension;
+            SimTK::Pathname::deconstructPathname(orientationsFileName,
+                    isAbsolutePath, directory, fileName, extension);
+            outName = "ik_" + fileName;
+        }
+        std::string outputFile = outName;
 
-    IO::makeDir(get_results_directory());
-    std::string outName = get_output_motion_file();
-    if (outName.empty()) { 
-        auto eix = orientationsFileName.rfind(".");
-        auto stix = orientationsFileName.rfind("/") + 1;
-        outName = "ik_" + orientationsFileName.substr(stix, eix - stix);
-    }
-    std::string outputFile = get_results_directory() + "/" + outName;
+        // Convert to degrees to compare with marker-based IK
+        // but only for rotational coordinates
+        model.getSimbodyEngine().convertRadiansToDegrees(report);
+        report.updTableMetaData().setValueForKey<string>("name", outName);
 
-    // Convert to degrees to compare with marker-based IK
-    // but only for rotational coordinates
-    model.getSimbodyEngine().convertRadiansToDegrees(report);
-    report.updTableMetaData().setValueForKey<string>("name", outName);
+        auto fullOutputFilename = outputFile;
+        std::string::size_type extSep = fullOutputFilename.rfind(".");
+        if (extSep == std::string::npos) { fullOutputFilename.append(".mot"); }
+        STOFileAdapter_<double>::write(report, fullOutputFilename);
 
-    auto fullOutputFilename = outputFile;
-    std::string::size_type extSep = fullOutputFilename.rfind(".");
-    if (extSep == std::string::npos) { 
-        fullOutputFilename.append(".mot");
-    }
-    STOFileAdapter_<double>::write(report, fullOutputFilename);
-
-    log_info(
-            "Wrote IK with IMU tracking results to: '{}'.", fullOutputFilename);
-    if (get_report_errors()) {
-        STOFileAdapter_<double>::write(*modelOrientationErrors,
-                get_results_directory() + "/" +
-                        getName() + "_orientationErrors.sto");
-    }
+        log_info("Wrote IK with IMU tracking results to: '{}'.",
+                fullOutputFilename);
+        if (get_report_errors()) {
+            STOFileAdapter_<double>::write(*modelOrientationErrors,
+                    getName() + "_orientationErrors.sto");
+        }
+    } 
+    else
+        log_info("IMUInverseKinematicsTool: No output files are written.");
```

Comment:
Should this case actually just throw an Exception instead of logging that no files were written?

I think this will also be called if both `results_dir` and `output_motion_file` are used? It may be useful to have a separate message for this.

If there's a use case for this (e.g., running this without outputting files) I'd just add one more statement here to give the user a reason why files were written so that they can fix it if it's a problem. Perhaps "Either the results directory or output_file_name must be specified to write to file."

Reply:
Thanks for the review @carmichaelong The use case I ran into was running from the GUI with neither output files nor results_directory specified, so results are displayed in GUI. Despite that, spurious  .mot files were created in random places. I agree there's a potential for inconsistency (e.g. results_dir specified while output file has a full path specified). In this case the code gives precedence to results_dir though a warning may be justified.

Reply:
> The use case I ran into was running from the GUI with neither output files nor results_directory specified, so results are displayed in GUI.

Ah, I see, thanks for the clarification. I'd say then at least adding a message to the `log_info` line to indicate how to get output files written would be great to help out users. Otherwise this PR looks good to me

Reply:
I did some tests in MATLAB. If I don't specify `set_results_directory` but do `imuIK.set_output_motion_file('IKResults/testIK')` then the results are saved in the current folder, ie not in a new folder IKResults. Is that what we want?

The rest of the tests went fine.